src/main/scala/org/ergoplatform/modifiers/mempool/UnconfirmedTransaction.scala

```diff
@@ -0,0 +1,60 @@
+package org.ergoplatform.modifiers.mempool
+
+import scorex.core.serialization.ScorexSerializer
+import scorex.util.{ModifierId, ScorexLogging}
+import scorex.util.serialization.{Reader, Writer}
+
+
+case class UnconfirmedTransaction(transaction: ErgoTransaction,
+                                  lastCost: Option[Int],
+                                  createdTime: Long,
+                                  lastCheckedTime: Long,
+                                  transactionBytes: Option[Array[Byte]])
+  extends ScorexLogging {
+
+  def id: ModifierId = transaction.id
+
+  def updateCost(cost: Int): UnconfirmedTransaction = {
+    copy(lastCost = Some(cost), lastCheckedTime = System.currentTimeMillis())
+  }
+
+}
+
+object UnconfirmedTransaction {
+
+  def apply(tx: ErgoTransaction): UnconfirmedTransaction = UnconfirmedTransaction(tx, None, 0, 0, None)
+
+  def apply(tx: ErgoTransaction, txBytes: Array[Byte]): UnconfirmedTransaction = {
+    UnconfirmedTransaction(tx, None, 0, 0, Some(txBytes))
+  }
+
+}
+
+object UnconfirmedTransactionSerializer extends ScorexSerializer[UnconfirmedTransaction] {
+
+  override def serialize(unconfirmedTx: UnconfirmedTransaction, w: Writer): Unit = {
+    ErgoTransactionSerializer.serialize(unconfirmedTx.transaction, w)
+    w.putOption(unconfirmedTx.lastCost)(_.putUInt(_))
+    w.putULong(unconfirmedTx.createdTime)
+    w.putULong(unconfirmedTx.lastCheckedTime)
+    w.putOption(unconfirmedTx.transactionBytes){(_,d) =>
+      w.putInt(d.length)
```

Comment:
better use unsigned encoding of Int (putUInt), it is more efficient

Reply:
removed unused serializer